pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1426,7 +1432,8 @@ def visit_assign(self, node: nodes.Assign) -> None:
         "consider-using-ternary",
         "consider-swap-variables",
     )
-    def visit_return(self, node: nodes.Return) -> None:
+    def visit_return(self, node: nodes.Return | nodes.Assign) -> None:
+        """This function can be called from 'visit_assign'."""
```

Comment:
The typing would be correct. However, I'm not really a fan of calling `visit_return` from another `visit_*` method. Maybe it would make sense to create a new method for it and call that one instead. Something like `_check_simplify_expression`? Anyway this should probably be done separately.

Reply:
I agree, it's better to have a subfunction called from both visit functions and proper unsurprising typing on the visit function. There are other instance of that, in the codebase I could open another merge request, searching for ``| nodes.`` should work I guess.

Reply:
https://github.com/PyCQA/pylint/pull/6892